include/zenohcxx/api.hxx

```diff
@@ -670,6 +670,153 @@ struct Timestamp : Copyable<::z_timestamp_t> {
 };
 
 #ifdef __ZENOHCXX_ZENOHC
+/// The attachment vtable.
+struct AttachmentVTable : public Copyable<::z_attachment_vtable_t> {
+    using Copyable::Copyable;
+
+    /// @name Constructors
+
+    /// @brief AttachmentVTable with a iteration driver and a len functions
+    AttachmentVTable(z_attachment_iter_driver_t _iter_driver, z_attachment_len_t _len_func)
+        : Copyable({_iter_driver, _len_func}) {}
+
+    /// @name Methods
+
+    /// @brief Set the iteration driver
+    /// @param _iter_driver value of ``z_attachment_iter_driver_t`` type
+    /// @return Reference to the ``AttachmentVTable`` object
+    AttachmentVTable& set_iteration_driver(z_attachment_iter_driver_t _iter_driver) {
+        iteration_driver = _iter_driver;
+        return *this;
+    }
+
+    /// @brief Set the len function
+    /// @param _len_func value of ``z_attachment_len_t`` type
+    /// @return Reference to the ``AttachmentVTable`` object
+    AttachmentVTable& set_len_func(z_attachment_len_t _len_func) {
+        len = _len_func;
+        return *this;
+    }
+
+    /// @brief Get the iteration driver
+    /// @return value of ``z_attachment_iter_driver_t`` type
+    z_attachment_iter_driver_t get_iteration_driver() const { return iteration_driver; }
+
+    /// @brief Get the len function
+    /// @return value of ``z_attachment_len_t`` type
+    z_attachment_len_t get_len_func() const { return len; }
+
+    /// @name Operators
+
+    /// @brief Equality operator
+    /// @param v other ``AttachmentVTable`` object
+    /// @return true if the attachment v-tables are equal
+    bool operator==(const AttachmentVTable& v) const {
+        // TODO(sashacmc): should we implement a deep comparing?
+        return get_iteration_driver() == v.get_iteration_driver() && get_len_func() == v.get_len_func();
+    }
+
+    /// @brief Inequality operator
+    /// @param v other ``AttachmentVTable`` object
+    /// @return true if the attachment v-tables are not equal
+    bool operator!=(const AttachmentVTable& v) const { return !operator==(v); }
+};
+
+/// The attachment.
+/// A v-table based map of byte slice to byte slice.
+struct Attachment : public Copyable<::z_attachment_t> {
```

Comment:
@milyin We might indeed consider it if we realize we need to expose an owned variant of attachments for replies. For any other usage, we never want to take nor give ownership of the attachments, so it would be unnecessary to add it.